### PR TITLE
Cell level from a length scale

### PR DESCRIPTION
**Pull request by svenkreiss**
* implement `get_min_level()` and `get_closest_level()`
* introduce `kMaxAngleSpan`, `kAvgEdge`, `kMinEdge`, `kMaxEdge`

### CODE REVIEW

**Comment by danvk** [comment]
Is this translated directly from the C++? Can you link to the relevant bits?

**Comment by svenkreiss** [comment]
Yes, the values for kAvgEdge and so on are from here: https://github.com/micolous/s2-geometry-library/blob/ad4673593fe23bbf8aacce4aead577db0bae9f5f/geometry/s2.cc#L728
`GetMinLevel()` and related are implemented here: https://github.com/micolous/s2-geometry-library/blob/ad4673593fe23bbf8aacce4aead577db0bae9f5f/geometry/s2/s2.h#L819

**Comment by svenkreiss** [comment]
Just added the equivalent instantiations for the `AreaMetric` to work with cell area.

File: s2sphere/sphere.py [hunk 1]
```diff
@@ -1538,10 +1538,19 @@ def get_value(self, level):
         return math.ldexp(self.deriv(), -self.__dim * level)
 
     def get_closest_level(self, value):
-        raise NotImplementedError()
+        return self.get_min_level(
+            (math.sqrt(2) if self.__dim == 1 else 2) * value
+        )
 
     def get_min_level(self, value):
```
Comment (danvk) [inline s2sphere/sphere.py]: Copy over the documentation while you're in here?
https://github.com/micolous/s2-geometry-library/blob/ad4673593fe23bbf8aacce4aead577db0bae9f5f/geometry/s2/s2.h#L472-L476

**Comment by danvk** [comment]
Copy over the docs, otherwise LGTM.

File: s2sphere/sphere.py [hunk 2]
```diff
@@ -1538,12 +1538,35 @@ def get_value(self, level):
         return math.ldexp(self.deriv(), -self.__dim * level)
 
     def get_closest_level(self, value):
-        raise NotImplementedError()
+        # Return the level at which the metric has approximately the given
```
Comment (danvk) [inline s2sphere/sphere.py]: DOCSTRINGS!!!!

**Pull request merged**